File: client/python/ws-publish-test.py
#!/usr/bin/env python3
"""
WebSocket Test Publisher - accepts JSON payload as command-line argument
Usage: python3 ws-publish-test.py '{"temp":72.5,"location":"test-lab"}'
"""

import sys
import json
import asyncio
import websockets

if len(sys.argv) < 2:
    print('Usage: python3 ws-publish-test.py \'{"temp":72.5,"location":"test-lab"}\'')
    sys.exit(1)

# Parse payload from command line
payload_data = json.loads(sys.argv[1])

async def publish():
    """Publish a single message via WebSocket"""
    uri = "ws://localhost:9926/Sensors/101"

    try:
        async with websockets.connect(uri) as websocket:
            print("Connected to Harper WebSocket")

            # WebSocket expects array format (matching ws-publish.js behavior)
            payload = json.dumps([payload_data])

            await websocket.send(payload)
            print(f"Published: {payload}")

    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

def main():
    asyncio.run(publish())

if __name__ == "__main__":
    main()

File: client/nodejs/ws-publish-test.js
#!/usr/bin/env node
const WebSocket = require('ws');

// Get payload from command line argument
if (process.argv.length < 3) {
  console.error('Usage: node ws-publish-test.js \'{"temp":72.5,"location":"test-lab"}\'');
  process.exit(1);
}

const payloadData = JSON.parse(process.argv[2]);
const ws = new WebSocket('ws://localhost:9926/Sensors/101');

ws.on('open', () => {
  console.log('Connected to Harper WebSocket');

  // WebSocket expects array format
  const payload = JSON.stringify([payloadData]);

  ws.send(payload, (err) => {
    if (err) {
      console.error('Error sending message:', err);
      process.exit(1);
    } else {
      console.log(`Published: ${payload}`);
      ws.close();
    }
  });
});

ws.on('error', (err) => {
  console.error('WebSocket error:', err.message);
  process.exit(1);
});

ws.on('close', () => {
  process.exit(0);
});
